rules.bzl

```diff
@@ -42,12 +42,17 @@ def _android_ndk_repository_impl(ctx):
 
     api_level = ctx.attr.api_level or 31
 
-    lib64_clang_directory = clang_directory + "/lib64/clang"
-    lib64_clang_files = ctx.path(lib64_clang_directory).readdir()
-    if len(lib64_clang_files) != 1:
-        fail("Expected 1 directory under " + lib64_clang_directory)
-    clang_version = lib64_clang_files[0].basename
-    clang_resource_directory = "lib64/clang/%s" % clang_version
+    lib_dir = "lib64"
+    lib_clang_directory = ctx.path(clang_directory + "/{}/clang".format(lib_dir))
+    if not lib_clang_directory.exists:
+        lib_dir = "lib"
+        lib_clang_directory = ctx.path(clang_directory + "/{}/clang".format(lib_dir))
+    if not lib_clang_directory.exists:
+        fail("Expected directory %s to exist" % lib_clang_directory)
+
+    # If the directory contains '17' and '17.0.2' we want to take the latter.
+    clang_version = sorted(lib_clang_directory.readdir(), key = lambda x: x.basename)[-1].basename
+    clang_resource_directory = "lib/clang/%s" % clang_version
```

Comment:
looks like this needs to be reconciled with https://github.com/bazelbuild/rules_android_ndk/pull/42

should the changes here be reverted and #42 should be used?

Reply:
rebased!